### PR TITLE
Add metal llvm tools

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/metal-llvm-tools/conda_build_config.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+MACOSX_DEPLOYMENT_TARGET:  # [osx and x86_64]
+- 10.10                    # [osx and x86_64]
+MACOSX_SDK_VERSION:        # [osx and x86_64]
+  - 10.12                  # [osx and x86_64]
```
Comment (xhochy) [inline recipes/metal-llvm-tools/conda_build_config.yaml:4]: Either here or two lines above you need to adjust the whitespace.

File: recipes/metal-llvm-tools/build.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      -DLLVM_INCLUDE_BENCHMARKS=OFF \
+      -DLLVM_INCLUDE_DOCS=OFF \
+      -DLLVM_INCLUDE_EXAMPLES=OFF \
+      -DLLVM_INCLUDE_GO_TESTS=OFF \
+      -DLLVM_INCLUDE_TESTS=OFF \
+      -DLLVM_INCLUDE_UTILS=ON \
+      -DLLVM_INSTALL_UTILS=ON \
+      -DLLVM_UTILS_INSTALL_DIR=libexec/llvm \
+      -DLLVM_BUILD_LLVM_DYLIB=no \
+      -DLLVM_LINK_LLVM_DYLIB=no \
```
Comment (xhochy) [inline recipes/metal-llvm-tools/build.sh:40]: Does this imply static linkage to libllvm as we're building a custom version of it?

**Review by xhochy — APPROVED** [review]
Looking good, just a single question for my better understanding of it. Feel free to merge though.

File: recipes/metal-llvm-tools/build.sh (hunk 2 above)
Comment (isuruf) [reply recipes/metal-llvm-tools/build.sh:40]: Yes, static linking. This package outputs only two executables.

**Comment by isuruf** [comment]
Thanks for the review

**Pull request merged**